client/app/components/queries/editor-components/databricks/DatabricksSchemaBrowser.jsx

```diff
@@ -0,0 +1,147 @@
+import { useDebouncedCallback } from "use-debounce";
+import Input from "antd/lib/input";
+import Select from "antd/lib/select";
+import { SchemaList, applyFilterOnSchema } from "@/components/queries/SchemaBrowser";
+import useDatabricksSchema from "./useDatabricksSchema";
+
+import "./DatabricksSchemaBrowser.less";
+
+// Limit number of rendered options to improve performance until Antd v4
+function getLimitedDatabases(databases, currentDatabaseName, limit = 1000) {
+  const limitedDatabases = slice(without(databases, currentDatabaseName), 0, limit);
+
+  return currentDatabaseName ? [...limitedDatabases, currentDatabaseName].sort() : limitedDatabases;
+}
+
+export default function DatabricksSchemaBrowser({
+  dataSource,
+  options,
+  onOptionsUpdate,
+  onSchemaUpdate,
+  onItemSelect,
+  ...props
+}) {
+  const {
+    databases,
+    loadingDatabases,
+    schema,
+    loadingSchema,
+    currentDatabaseName,
+    setCurrentDatabase,
+  } = useDatabricksSchema(dataSource, options, onOptionsUpdate);
+  const [filterString, setFilterString] = useState("");
+  const [databaseFilterString, setDatabaseFilterString] = useState("");
+  const filteredSchema = useMemo(() => applyFilterOnSchema(schema, filterString), [schema, filterString]);
+  const [isDatabaseSelectOpen, setIsDatabaseSelectOpen] = useState(false);
+  const [expandedFlags, setExpandedFlags] = useState({});
+  const [handleFilterChange] = useDebouncedCallback(setFilterString, 500);
+  const [handleDatabaseFilterChange, cancelHandleDatabaseFilterChange] = useDebouncedCallback(
+    setDatabaseFilterString,
+    500
+  );
+
+  const handleDatabaseSelection = useCallback(
+    databaseName => {
+      setCurrentDatabase(databaseName);
+      cancelHandleDatabaseFilterChange();
+      setDatabaseFilterString("");
+    },
+    [cancelHandleDatabaseFilterChange, setCurrentDatabase]
+  );
+
+  const filteredDatabases = useMemo(
+    () => filter(databases, database => includes(database.toLowerCase(), databaseFilterString.toLowerCase())),
+    [databases, databaseFilterString]
+  );
+  const limitedDatabases = useMemo(() => getLimitedDatabases(filteredDatabases, currentDatabaseName), [
+    filteredDatabases,
+    currentDatabaseName,
+  ]);
+
+  const onSchemaUpdateRef = useRef();
+  onSchemaUpdateRef.current = onSchemaUpdate;
+  useEffect(() => {
+    setExpandedFlags({});
+    if (isFunction(onSchemaUpdateRef.current)) {
+      onSchemaUpdateRef.current(schema);
+    }
+  }, [schema]);
+
+  if (schema.length === 0 && databases.length === 0 && !(loadingDatabases || loadingSchema)) {
+    return null;
+  }
+
+  function toggleTable(tableName) {
+    setExpandedFlags({
+      ...expandedFlags,
+      [tableName]: !expandedFlags[tableName],
+    });
+  }
+
+  return (
+    <div className="databricks-schema-browser schema-container" {...props}>
+      <div className="schema-control">
+        <Input
+          className={isDatabaseSelectOpen ? "database-select-open" : ""}
```

Comment:
So, @arikfr, how badly do you want the Select to be white when open? 😆

Turns out the grey bg comes from the input addon wrapper, recreating the borders from the Select component feels too much and wasn't that simple. `:has()` is the css pseudo-selector we could apply here, but will be avail in CSS4, so, "simplest" option was to use JS to achieve that.

The result so far is this:
<img width="238" alt="Screen Shot 2020-06-29 at 21 56 32" src="https://user-images.githubusercontent.com/3356951/86070575-7539de00-ba53-11ea-8c2e-eebd305922f9.png">
 
Can still remove that placeholder from current DB if you think it's relevant, although I think it makes sense to be there, so we don't get the feeling that the option was lost.

Reply:
Writing on top of the previous value feels like a bug... 

> so we don't get the feeling that the option was lost.

Can we open the list so the selected option is visible and there is a checkmark next to it?

Reply:
> Can we open the list so the selected option is visible and there is a checkmark next to it?

It's already like this. Anyway, done in https://github.com/getredash/redash/pull/5010/commits/69b79341af308512a848c897cd14365f4b0c01a9, let's give it a try.